lib/route.coffee

```diff
@@ -159,10 +159,10 @@ class @Route
     auth = @api.config.auth.user.call(endpointContext)
 
     # Get the user from the database
-    if not auth?.user and auth?.userId and auth?.token
+    if auth?.userId and auth?.token and not auth?.user
       userSelector = {}
       userSelector._id = auth.userId
-      userSelector[@api.config.auth.token] = auth.token
+      userSelector[@api.config.auth.token] = Accounts._hashLoginToken auth.token
```

Comment:
That sounds fine to me. Of course, as long as the unit tests pass! Lol
On Jun 17, 2015 6:01 PM, "Kahmali Rose" <notifications@github.com> wrote:

> In lib/route.coffee
> <https://github.com/kahmali/meteor-restivus/pull/82#discussion_r32684789>:
>
> >        userSelector = {}
> >        userSelector._id = auth.userId
> > -      userSelector[@api.config.auth.token] = auth.token
> > +      userSelector[@api.config.auth.token] = Accounts._hashLoginToken auth.token
>
> Almost missed this:
>
> Hashing the token should actually be done from the default auth function.
> This way it doesn't interfere with the custom auth discussed in #6
> <https://github.com/kahmali/meteor-restivus/issues/6>, and it will also
> play nicely with #49
> <https://github.com/kahmali/meteor-restivus/issues/49>. Otherwise, users
> with custom auth would be forced to hash their tokens the same way, which
> doesn't provide enough flexibility. I'll fix this myself, but I just wanted
> to run it by you in case you had any objections.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kahmali/meteor-restivus/pull/82/files#r32684789>.
>